node/src/service.rs

```diff
@@ -191,21 +270,41 @@ pub fn new_partial(
 	})
 }
 
-pub struct NewFullBase {
+pub struct NewFull<C> {
 	pub task_manager: TaskManager,
-	pub client: Arc<FullClient>,
+	pub client: C,
 	pub network: Arc<NetworkService<Block, <Block as BlockT>::Hash>>,
-	pub transaction_pool: Arc<sc_transaction_pool::FullPool<Block, FullClient>>,
+}
+
+impl<C> NewFull<C> {
+	/// Convert the client type using the given `func`.
+	pub fn with_client<NC>(self, func: impl FnOnce(C) -> NC) -> NewFull<NC> {
+		NewFull {
+			task_manager: self.task_manager,
+			client: func(self.client),
+			network: self.network,
+		}
+	}
+}
+
+pub fn build_full(config: Configuration, run_testing_runtime: bool) -> Result<NewFull<Client>, ServiceError> {
+	if run_testing_runtime {
+		new_full::<testing_hydra_dx_runtime::RuntimeApi, TestingHydraExecutor>(config)
+			.map(|full| full.with_client(Client::TestingHydraDX))
+	} else {
+		new_full(config).map(|full| full.with_client(Client::HydraDX))
+	}
 }
 
 /// Creates a full service from the configuration.
-pub fn new_full_base(
+pub fn new_full<RuntimeApi, Executor>(
```

Comment:
can we use this pattern in client.rs?

Reply:
Could you be please more specific where to use the pattern? I'm not sure.

Reply:
I was thinking about making client configurable instead of copying stuff but when looking at it it might not be an option.